ts/src/pro/blofin.ts

```diff
@@ -0,0 +1,489 @@
+        //               "0.08179",
+        //               "0.08194",
+        //               "7695",
+        //               "7695000",
+        //               "630099.32",
+        //               "0",
+        //             ],
+        //         ],
+        //     }
+        //
+        const arg = this.safeDict (message, 'arg');
+        const channelName = this.safeString (arg, 'channel');
+        const data = this.safeList (message, 'data');
+        const marketId = this.safeString (arg, 'instId');
+        const market = this.safeMarket (marketId);
+        const symbol = market['symbol'];
+        const interval = channelName.replace ('candle', '');
+        const unifiedTimeframe = this.findTimeframe (interval);
+        this.ohlcvs[symbol] = this.safeValue (this.ohlcvs, symbol, {});
+        let stored = this.safeValue (this.ohlcvs[symbol], unifiedTimeframe);
+        if (stored === undefined) {
+            const limit = this.safeInteger (this.options, 'OHLCVLimit', 1000);
+            stored = new ArrayCacheByTimestamp (limit);
+            this.ohlcvs[symbol][unifiedTimeframe] = stored;
+        }
+        for (let i = 0; i < data.length; i++) {
+            const candle = data[i];
+            const parsed = this.parseOHLCV (candle, market);
+            stored.append (parsed);
+        }
+        const resolveData = [ symbol, unifiedTimeframe, stored ];
+        const messageHash = 'candle' + interval + ':' + symbol;
+        client.resolve (resolveData, messageHash);
+    }
+
+    handleMessage (client: Client, message) {
+        //
+        // message examples
+        //
+        // {
+        //   arg: {
+        //     channel: "trades",
+        //     instId: "DOGE-USDT",
+        //   },
+        //   event: "subscribe"
+        // }
+        //
+        // incoming data updates' examples can be seen under each handler method
+        //
+        const methods = {
+            'trades': this.handleWsTrades,
+            'books': this.handleWsOrderBook,
+            'tickers': this.handleWsTicker,
+            'candle': this.handleWsOHLCV,
+            'pong': this.handleWsPong,
+        };
+        let method = undefined;
+        if (message === 'pong') {
+            method = this.safeValue (methods, 'pong');
+        } else {
+            const event = this.safeString (message, 'event');
+            if (event === 'subscribe') {
+                return;
+            }
+            const arg = this.safeDict (message, 'arg');
+            const channelName = this.safeString (arg, 'channel');
+            method = this.safeValue (methods, channelName);
+            if (!method && channelName.indexOf ('candle') >= 0) {
+                method = methods['candle'];
+            }
+        }
+        if (method) {
+            method.call (this, client, message);
+        }
+    }
+
+    async watchMultipleWrapper (channelName: string, callerMethodName: string, symbolsArray, params = {}) {
```

Comment:
we can type symbolsArray here

Reply:
it can be either flat array of symbols:
`[ 'BTC/USDT', 'ETH/USDT']`
or array of symbol & timeframes:
`[ ['BTC/USDT', '1m'], ['ETH/USDT', '5m'] ]`.

so, we can type it as generic array, lmk if last commit looks ok.